src/mod2c_core/nocpout.c

```diff
@@ -127,6 +127,7 @@ not thread safe and _p and _ppvar are static.
 #define NRNPOINTER	04000
 #define IONCONC		010000
 #define NRNBBCOREPOINTER	020000
+#define IONCONC_IMPLICIT 040000
```

Comment:
Today then calls to `nrn_wrote_conc` assume that the different ion variables for a given instance are adjacent in memory and can therefore be found via pointer arithmetic. The point of these extra explicit ion variables was to avoid that assumption. This was **necessary** for the "pure SoA" version of this work.

Now that we have SoAoS support, for array variables, we could in principle use that layout for ion data too, which would allow the old pointer arithmetic method to still work. This has not been done, and would clearly have implications for the data access pattern in other parts of the ion handling code.